core/spec/models/spree/shipment_spec.rb

```diff
@@ -476,38 +486,78 @@
 
     context "when order is not paid" do
       before { allow(order).to receive_messages(paid?: false) }
-      it "should result in a 'ready' state" do
+      it "should result in a 'pending' state" do
         shipment.resume!
         expect(shipment.state).to eq 'pending'
       end
     end
 
     context "when any inventory is backordered" do
-      before { allow_any_instance_of(Spree::InventoryUnit).to receive(:allow_ship?).and_return(false) }
-      it "should result in a 'ready' state" do
+      before { allow(shipment.stock_location).to receive(:fill_status).and_return([0, 1]) }
+      it "should result in a 'pending' state" do
         shipment.resume!
         expect(shipment.state).to eq 'pending'
       end
     end
 
+    context "when any inventory was already canceled" do
```

Comment:
Thanks for adding this spec!

Reply:
Thanks to you for pointing out the edge case 🙇, I totally missed it!